src/twisted/trial/_dist/test/test_workerreporter.py

```diff
@@ -5,129 +5,119 @@
 Tests for L{twisted.trial._dist.workerreporter}.
 """
 
-from twisted.python.failure import Failure
-from twisted.trial._dist import managercommands
-from twisted.trial._dist.workerreporter import WorkerReporter
-from twisted.trial.unittest import TestCase, Todo
 
+from unittest import TestCase
 
-class FakeAMProtocol:
+from hamcrest import assert_that, equal_to, has_length, has_properties
+from hamcrest.core.matcher import Matcher
+
+from twisted.test.iosim import connectedServerAndClient
+from twisted.trial._dist.worker import LocalWorkerAMP, WorkerProtocol
+from twisted.trial.reporter import TestResult
+from twisted.trial.test import erroneous, pyunitcases, sample, skipping
+from twisted.trial.unittest import SynchronousTestCase
+
+
+def matches_result(
+    successes=equal_to(0),
+    errors=has_length(0),
+    failures=has_length(0),
+    skips=has_length(0),
+    expectedFailures=has_length(0),
+    unexpectedSuccesses=has_length(0),
+):
     """
-    A fake C{AMP} implementations to track C{callRemote} calls.
+    Match a L{TestCase} instances with matching attributes.
     """
+    return has_properties(
+        {
+            "successes": successes,
+            "errors": errors,
+            "failures": failures,
+            "skips": skips,
+            "expectedFailures": expectedFailures,
+            "unexpectedSuccesses": unexpectedSuccesses,
+        }
+    )
+
+
+def run(case: SynchronousTestCase, target: TestCase) -> TestResult:
+    """
+    Run C{target} and return a test result as populated by a worker reporter.
 
-    id = 0
-    lastCall = None
-
-    def callRemote(self, command, **kwargs):
-        self.lastCall = command
-        self.lastArgs = kwargs
+    @param case: A test case to use to help run the target.
+    """
+    result = TestResult()
+    worker, local, pump = connectedServerAndClient(LocalWorkerAMP, WorkerProtocol)
+    d = local.run(target, result)
+    pump.pump()
+    pump.pump()
+    assert_that(case.successResultOf(d), equal_to({"success": True}))
+    return result
 
 
-class WorkerReporterTests(TestCase):
+class WorkerReporterTests(SynchronousTestCase):
     """
     Tests for L{WorkerReporter}.
     """
 
-    def setUp(self):
-        self.fakeAMProtocol = FakeAMProtocol()
-        self.workerReporter = WorkerReporter(self.fakeAMProtocol)
-        self.test = TestCase()
-
-    def test_addSuccess(self):
+    def _check(self, target: TestCase, **expectations: Matcher) -> None:
```

Comment:
Check is kind of generic.


Do we need it to be a private method?

Maybe

```suggestion
    def assertTestRun(self, target: TestCase, **expectations: Matcher) -> None:
```